server/libs/infra/src/infra.config.ts

```diff
@@ -3,17 +3,34 @@ import { BullModuleOptions } from '@nestjs/bull';
 import { RedisOptions } from 'ioredis';
 import { ConfigurationOptions } from 'typesense/lib/Typesense/Configuration';
 
-export const redisConfig: RedisOptions = {
-  host: process.env.REDIS_HOSTNAME || 'immich_redis',
-  port: parseInt(process.env.REDIS_PORT || '6379'),
-  db: parseInt(process.env.REDIS_DBINDEX || '0'),
-  password: process.env.REDIS_PASSWORD || undefined,
-  path: process.env.REDIS_SOCKET || undefined,
-};
+export function redisConfig(): RedisOptions {
+  const redisUrl = process.env.REDIS_URL;
+  if (redisUrl != undefined && redisUrl.startsWith('ioredis://')) {
+    let customOptions = {};
+    try {
+      const decodedString = Buffer.from(redisUrl.slice(10), 'base64').toString();
+      customOptions = JSON.parse(decodedString);
+    } catch (error) {
+      throw new Error(`Failed to decode redis options: ${error}`);
+    }
+    return customOptions;
+  } else {
```

Comment:
If you return from all paths in the above condition you don't need the else condition explicitly anymore.